cranker/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+                    &spl_token::id()
+                );
+
+                let recent_blockhash = self.rpc_client.get_latest_blockhash()?;
+                let create_ata_tx = Transaction::new_signed_with_payer(
+                    &[create_ata_ix],
+                    Some(&self.payer.pubkey()),
+                    &[self.payer.as_ref()],
+                    recent_blockhash
+                );
+
+                self.rpc_client.send_and_confirm_transaction(&create_ata_tx)?;
+                info!("Created fee wallet token account");
+            }
+        }
+
+        // Now proceed with claim
+        info!("Creating claim instruction after verification");       
+
+        let claim_ix = create_claim_pool_tokens_instruction(
+            &self.program_id,
+            &receipt.base,
+            &receipt.owner,
+            &stake_pool_deposit_authority.vault,
+            &owner_ata,
+            &fee_wallet_token_account, // Changed from stake_pool_deposit_authority.fee_wallet
+            &receipt.stake_pool_deposit_stake_authority,
+            &stake_pool_deposit_authority.pool_mint,
+            &spl_token::id(),
+            true
+        );
+
+        let recent_blockhash = self.rpc_client.get_latest_blockhash()?;
+        let claim_tx = Transaction::new_signed_with_payer(
+            &[claim_ix],
+            Some(&self.payer.pubkey()),
+            &[&*self.payer],
+            recent_blockhash
+        );
+
+        match self.rpc_client.send_and_confirm_transaction(&claim_tx) {
+            Ok(sig) => {
+                info!(
+                    "Successfully claimed pool tokens for receipt {}. Transaction signature: {}",
+                    receipt.base,
+                    sig
+                );
+                Ok(())
+            }
+            Err(e) => {
+                error!("Failed to claim pool tokens for receipt {}. Error: {}", receipt.base, e);
+                Err(CrankerError::RpcError(e))
+            }
+        }
+    }
+
+    async fn get_stake_pool_deposit_authority(
+        &self,
+        pubkey: &Pubkey
+    ) -> Result<StakePoolDepositStakeAuthority, CrankerError> {
+        let account = self.rpc_client.get_account(pubkey).map_err(CrankerError::RpcError)?;
+
+        // Skip 8-byte discriminator
+        if account.data.len() < 8 + std::mem::size_of::<StakePoolDepositStakeAuthority>() {
+            return Err(
+                CrankerError::DeserializeError(
+                    format!(
+                        "Account data too short: {}, expected: {}",
+                        account.data.len(),
+                        8 + std::mem::size_of::<StakePoolDepositStakeAuthority>()
+                    )
+                )
+            );
+        }
+
+        bytemuck
+            ::try_from_bytes::<StakePoolDepositStakeAuthority>(&account.data[8..])
+            .map(|auth| *auth)
+            .map_err(|e| CrankerError::DeserializeError(e.to_string()))
```

Comment:
Let's use `StakePoolDepositStakeAuthority::try_from_slice_unchecked` to simplify.